### ISSUE TITLE
FairMQ could not be created with latest dev version

### ISSUE THREAD

**Issue by AliMRamos** [FairRootGroup/FairSoft#329]
Dear developers,

I am on the following dev version of FairSoft: [995dafd](https://github.com/FairRootGroup/FairSoft/commit/995dafd399172ec168150b5f5c8a8c45e16f70de)
on macOS 10.15.6 Catalina
Xcode version 12 (12A7209)

During the installation process I get the following error:

```
Cloning into 'FairMQ'...
remote: Enumerating objects: 95, done.
remote: Counting objects: 100% (95/95), done.
remote: Compressing objects: 100% (73/73), done.
remote: Total 18009 (delta 44), reused 35 (delta 22), pack-reused 17914
Receiving objects: 100% (18009/18009), 15.97 MiB | 4.28 MiB/s, done.
Resolving deltas: 100% (12863/12863), done.
*** Compiling FairMQ ................ 
Note: switching to 'v1.4.8'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 3f537482 CMake: Add config for --graphviz
-- Found Git: /usr/bin/git (found version "2.24.3 (Apple Git-128)") 
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- FairMQ 1.4.8 from Wed Sep 4 21:17:35 2019 +0200
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
--   
--   CXX STANDARD       C++11 (>= C++11, change with -DCMAKE_CXX_STANDARD=17)
--   
--   GLOBAL CXX FLAGS  -O2 -std=c++11 -stdlib=libc++
--   
--   BUILD TYPE         CXX FLAGS
--   Debug              -g -Wshadow -Wall -Wextra
--   Release            -O2 -DNDEBUG
-- * RelWithDebInfo     -O2 -g -Wshadow -Wall -Wextra -DNDEBUG
--   Nightly            -O2 -g -Wshadow -Wall -Wextra
--   Profile            -g3 -Wshadow -Wall -Wextra -fno-inline -ftest-coverage -fprofile-arcs
--   Experimental       -O2 -g -Wshadow -Wall -Wextra -DNDEBUG
--   AdressSan          -O2 -g -Wshadow -Wall -Wextra -fsanitize=address -fno-omit-frame-pointer
--   ThreadSan          -O2 -g -Wshadow -Wall -Wextra -fsanitize=thread
--   
--   (Change the build type with -DCMAKE_BUILD_TYPE=...)
--   
--   DEPENDENCY FOUND     VERSION                   PREFIX
--   Boost                1.68.0 (>= 1.67)          /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20
--   DDS                  2.4 (>= 2.4)              /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20
--   FairLogger           1.4.0 (>= 1.2.0)          /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20
--   ZeroMQ               4.3.1 (>= 4.1.5)          /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20
--   msgpack              3.1.1 (>= 3.1.0)          /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20
--   nanomsg              1.1.5 (>= 1.1.3)          /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20
--   
--   COMPONENT         BUILT?  INFO
--   fairmq             YES    (default, disable with -DBUILD_FAIRMQ=OFF)
--   tests               NO    (enable with -DBUILD_TESTING=ON)
--   nanomsg_transport  YES    (disable with -DBUILD_NANOMSG_TRANSPORT=OFF)
--   ofi_transport       NO    EXPERIMENTAL (requires C++14) (default, enable with -DBUILD_OFI_TRANSPORT=ON)
--   dds_plugin         YES    (disable with -DBUILD_DDS_PLUGIN=OFF)
--   pmix_plugin         NO    (default, enable with -DBUILD_PMIX_PLUGIN=ON)
--   examples           YES    (default, disable with -DBUILD_EXAMPLES=OFF)
--   docs                NO    (default, enable with -DBUILD_DOCS=ON)
--   sdk                 NO    EXPERIMENTAL (required C++14) (default, enable with -DBUILD_SDK=ON)
--   
--   INSTALL PREFIX     /Users/alicia/CODES/FAIRSOFT/fairsoft_sep20 (change with -DCMAKE_INSTALL_PREFIX=...)
--   
--   RUN STATIC ANALYSIS  OFF  (default, enable with -DRUN_STATIC_ANALYSIS=ON)
--   
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/build
Scanning dependencies of target fairmq-shmmonitor
Scanning dependencies of target Tools
[  0%] Building CXX object fairmq/CMakeFiles/fairmq-shmmonitor.dir/shmem/Monitor.cxx.o
[  1%] Building CXX object fairmq/CMakeFiles/fairmq-shmmonitor.dir/shmem/runMonitor.cxx.o
[  2%] Building CXX object fairmq/CMakeFiles/Tools.dir/tools/Network.cxx.o
[  3%] Building CXX object fairmq/CMakeFiles/Tools.dir/tools/Process.cxx.o
[  3%] Building CXX object fairmq/CMakeFiles/Tools.dir/tools/Semaphore.cxx.o
[  4%] Building CXX object fairmq/CMakeFiles/Tools.dir/tools/Unique.cxx.o
[  5%] Linking CXX executable fairmq-shmmonitor
[  5%] Built target fairmq-shmmonitor
[  5%] Linking CXX shared library libFairMQTools.dylib
[  5%] Built target Tools
Scanning dependencies of target StateMachine
[  7%] Building CXX object fairmq/CMakeFiles/StateMachine.dir/StateMachine.cxx.o
[  7%] Building CXX object fairmq/CMakeFiles/StateMachine.dir/States.cxx.o
[  7%] Linking CXX shared library libFairMQStateMachine.dylib
[  7%] Built target StateMachine
Scanning dependencies of target FairMQ
[  7%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/DeviceRunner.cxx.o
[  8%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQChannel.cxx.o
[  9%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQPoller.cxx.o
[  9%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQMessage.cxx.o
[  9%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQDevice.cxx.o
[ 10%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQLogger.cxx.o
[ 11%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQSocket.cxx.o
[ 12%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/FairMQTransportFactory.cxx.o
[ 12%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/devices/FairMQBenchmarkSampler.cxx.o
[ 13%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/devices/FairMQMerger.cxx.o
[ 14%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/devices/FairMQMultiplier.cxx.o
[ 14%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/devices/FairMQProxy.cxx.o
/Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/FairMQChannel.cxx:526:25: warning: loop variable 'endpoint' of type 'const std::__1::basic_string<char>' creates a copy from type
      'const std::__1::basic_string<char>' [-Wrange-loop-analysis]
        for (const auto endpoint : endpoints) {
                        ^
/Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/FairMQChannel.cxx:526:14: note: use reference type 'const std::__1::basic_string<char> &' to prevent copying
        for (const auto endpoint : endpoints) {
             ^~~~~~~~~~~~~~~~~~~~~
                        &
In file included from /Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/FairMQDevice.cxx:9:
In file included from /Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/FairMQDevice.h:12:
In file included from /Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/StateMachine.h:12:
In file included from /Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/States.h:12:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string:504:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/string_view:175:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__string:57:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:643:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:1725:13: error: static_assert failed due to requirement '__is_cpp17_move_insertable<std::__1::allocator<FairMQChannel>, false>::value'
      "The specified type does not meet the requirements of Cpp17MoveInsertable"
            static_assert(__is_cpp17_move_insertable<allocator_type>::value,
            ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:952:21: note: in instantiation of function template specialization 'std::__1::allocator_traits<std::__1::allocator<FairMQChannel>
      >::__construct_backward_with_exception_guarantees<FairMQChannel *>' requested here
    __alloc_traits::__construct_backward_with_exception_guarantees(
                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:1668:5: note: in instantiation of member function 'std::__1::vector<FairMQChannel, std::__1::allocator<FairMQChannel>
      >::__swap_out_circular_buffer' requested here
    __swap_out_circular_buffer(__v);
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:1686:9: note: in instantiation of function template specialization 'std::__1::vector<FairMQChannel, std::__1::allocator<FairMQChannel>
      >::__emplace_back_slow_path<const std::__1::basic_string<char> &, int &, std::__1::map<std::__1::basic_string<char>, boost::any, std::__1::less<std::__1::basic_string<char> >,
      std::__1::allocator<std::__1::pair<const std::__1::basic_string<char>, boost::any> > > >' requested here
        __emplace_back_slow_path(_VSTD::forward<_Args>(__args)...);
        ^
/Users/alicia/CODES/FAIRSOFT/FairSoft/legacy/basics/FairMQ/fairmq/FairMQDevice.cxx:285:35: note: in instantiation of function template specialization 'std::__1::vector<FairMQChannel,
      std::__1::allocator<FairMQChannel> >::emplace_back<const std::__1::basic_string<char> &, int &, std::__1::map<std::__1::basic_string<char>, boost::any, std::__1::less<std::__1::basic_string<char> >,
      std::__1::allocator<std::__1::pair<const std::__1::basic_string<char>, boost::any> > > >' requested here
            fChannels[info.first].emplace_back(info.first, i, fConfig->GetPropertiesStartingWith(tools::ToString("chans.", info.first, ".", i, ".")));
                                  ^
1 error generated.
make[2]: *** [fairmq/CMakeFiles/FairMQ.dir/FairMQDevice.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 warning generated.
make[1]: *** [fairmq/CMakeFiles/FairMQ.dir/all] Error 2
make: *** [all] Error 2
*** ERROR: FairMQ could not be created.
*** End installation of external packages with Errors***
```

**Comment by dennisklein** [comment]
I believe I pushed a fix to the `dev` branch: fb2c4124946f35d27c13405baab516d4a42080c0. Can you please try again?

**Comment by AliMRamos** [comment]
Yes, issue fixed, thanks!
